src/abstracts/SablierV2MerkleLockup.sol

```diff
@@ -97,11 +100,12 @@ abstract contract SablierV2MerkleLockup is
 
     /// @inheritdoc ISablierV2MerkleLockup
     function clawback(address to, uint128 amount) external override onlyAdmin {
-        // Check: the campaign is not expired.
-        if (!hasExpired()) {
-            revert Errors.SablierV2MerkleLockup_CampaignNotExpired({
+        // Check: current timestamp is over the grace period and the campaign has not expired.
+        if (_firstClaimTime > 0 && block.timestamp > _firstClaimTime + 7 days && !hasExpired()) {
```

Comment:
what if we add a new function to increase the readability:

```solidity
    function _hasGracePeriodPassed() internal view returns (bool) {
        return _firstClaimTime > 0 && block.timestamp > _firstClaimTime + 7 days;
    }
    
     function clawback(address to, uint128 amount) external override onlyAdmin {
        if (_hasGracePeriodPassed() && !hasExpired()) {}
     }
```

should it be `public` or `internal`? `hasGracePeriodPassed` or `isWithinGracePeriod`?

Reply:
Agree. Separating it out in the form of a function would increase the readability.

- I would go with `internal`
- We can create a getter that returns `_firstClaimTime` that can be used to find out if its within grace period or not.
- `hasGracePeriodPassed` looks good to me. `isWithinGracePeriod` would require a different combination of conditions. How about `postGracePeriod`?

Reply:
> We can create a getter that returns _firstClaimTime that can be used to find out if its within grace period or not

or we make the variable `public`

Reply:
Yes we can. I was thinking from the pov of gas usage. Given that its called on each `claim`, a getter will make it less expensive because the variable can continue to be `internal`.

Reply:
fair point